src/uu/base32/src/base_common.rs

```diff
@@ -11,58 +11,120 @@ use std::fs::File;
-        )
-        .optopt(
-            "w",
-            "wrap",
-            "wrap encoded lines after COLS character (default 76, 0 to disable wrapping)",
-            "COLS",
-        )
-        .parse(args);
+static VERSION: &str = env!("CARGO_PKG_VERSION");
 
-    let line_wrap = matches.opt_str("wrap").map(|s| match s.parse() {
-        Ok(n) => n,
-        Err(e) => {
-            crash!(1, "invalid wrap size: ‘{}’: {}", s, e);
-        }
-    });
-    let ignore_garbage = matches.opt_present("ignore-garbage");
-    let decode = matches.opt_present("decode");
+pub mod options {
+    pub static DECODE: &str = "decode";
+    pub static WRAP: &str = "wrap";
+    pub static IGNORE_GARBAGE: &str = "ignore-garbage";
+    pub static FILE: &str = "file";
+}
 
-    if matches.free.len() > 1 {
-        show_usage_error!("extra operand ‘{}’", matches.free[0]);
-        return 1;
+struct Config {
+    decode: bool,
+    ignore_garbage: bool,
+    wrap_cols: Option<usize>,
+    to_read: Option<String>,
+}
+
+impl Config {
+    fn from(options: clap::ArgMatches) -> Config {
+        let file: Option<String> = match options.values_of(options::FILE) {
+            Some(mut values) => {
+                let name = values.next().unwrap();
+                if values.len() != 0 {
+                    crash!(3, "extra operand ‘{}’", name);
+                }
+
+                if name == "-" {
+                    None
+                } else {
+                    if !Path::exists(Path::new(name)) {
+                        crash!(2, "{}: No such file or directory", name);
+                    }
+                    Some(name.to_owned())
+                }
+            }
+            None => None,
+        };
+
+        let cols = match options.value_of(options::WRAP) {
+            Some(num) => match num.parse::<usize>() {
+                Ok(n) => Some(n),
+                Err(e) => {
+                    crash!(1, "invalid wrap size: ‘{}’: {}", num, e);
+                }
+            },
+            None => None,
+        };
+
+        Config {
+            decode: options.is_present(options::DECODE),
+            ignore_garbage: options.is_present(options::IGNORE_GARBAGE),
+            wrap_cols: cols,
+            to_read: file,
+        }
     }
+}
 
-    if matches.free.is_empty() || &matches.free[0][..] == "-" {
-        let stdin_raw = stdin();
-        handle_input(
-            &mut stdin_raw.lock(),
-            format,
-            line_wrap,
-            ignore_garbage,
-            decode,
-        );
-    } else {
-        let path = Path::new(matches.free[0].as_str());
-        let file_buf = safe_unwrap!(File::open(&path));
-        let mut input = BufReader::new(file_buf);
-        handle_input(&mut input, format, line_wrap, ignore_garbage, decode);
+pub fn execute(args: Vec<String>, _summary: &str, long_help: &str, usage: &str, format: Format) -> i32 {
+    let app = App::new(executable!())
```

Comment:
please move the clap declaration in uumain for consistency with other programs

Reply:
Oh I see. I misunderstood your original comment.

Reply:
I am not sure what you understood, sorry :(
please do like here:
https://github.com/uutils/coreutils/blob/master/src/uu/ls/src/ls.rs#L510-L517
sorry

Reply:
Okay. I understood "Move the argument parsing to the file where uumain is". Now, I understand that you meant "Instead of having the `execute` function, just move everything to the `uumain` function itself.

Reply:
Sorry, communication can be hard :)

Reply:
No worries! Thank you for your patience :)